drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/client/ConnectionFactory.java

```diff
@@ -41,21 +44,33 @@
     private final EventLoopGroup group;
     private final SslContextFactory sslContextFactory;
     private final ByteBufAllocator allocator;
+    private DriftNettyConnectionFactoryConfig connectionFactoryConfig;
 
-    ConnectionFactory(EventLoopGroup group, SslContextFactory sslContextFactory, ByteBufAllocator allocator)
+    ConnectionFactory(
+            EventLoopGroup group,
+            SslContextFactory sslContextFactory,
+            ByteBufAllocator allocator,
+            DriftNettyConnectionFactoryConfig connectionFactoryConfig)
```

Comment:
Since we only use this config to determine which class we should use as `socketChannelClass` in `ConnectionFactory`, maybe we should just pass `Class socketChannelClass`?

Reply:
I see in #17 it is used in more sophisticated way. But still I'm not sure if it is a good idea to pass the entire config inside.

Reply:
yup, we are passing in the config to get hold of more properties. There are some properties we could add there in the future to fine tune. I think there is not a lot of risk in this.